frontend/src/services/effects.ts

```diff
@@ -164,6 +183,23 @@ export function withEffects(store: Store) {
     }
   })
 
+  /**
+   * When the user checks/unchecks counties in the `<CountySelector />`, we
+   * update `serviceAreas` to the service areas in the selected counties.
```

Comment:
Are we re-asking for data of we unselect? Where is the best place to check if we are adding or removing data?

Reply:
A few things:

1. In this specific effect, there is no network request - we just read data from the `SERVICE_AREAS_BY_COUNTY_BY_STATE` constant
2. Some other effects (eg. [this one](https://github.com/bayesimpact/tds/pull/153/files/e30b732f40a5414f3188e7decdf0852496a29241#diff-46255d068003bf7992c955f28ad811b4L85)) do make network requests. We memoize all read API requests ([example](https://github.com/bayesimpact/tds/blob/f887d8c52fc81c511c4e3a730bcdf586ee5d01b9/frontend/src/services/api.ts#L80)), so checking + unchecking won't trigger extra requests there
3. There is another case: if a user checks 2 service areas we issue an API request for rep points. If the user then unchecks one of those service areas, we issue *a second request* for just the remaining point (because the request for 2 areas was memoized, but the request for just one of those areas was not yet memoized). https://github.com/bayesimpact/tds/issues/119 tracks improving this, so we don't issue the 2nd request.